fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpaceDirectory.java

```diff
@@ -413,16 +413,23 @@ protected KeySpacePath wrap(@Nonnull KeySpacePath path) {
                 ? CompletableFuture.completedFuture(new Subspace())
                 : listFrom.toSubspaceAsync(context);
 
-        final ScanProperties singleRowScan = scanProperties.with(props -> props.setReturnedRowLimit(1));
         return new LazyCursor<>(
                 fromSubspaceFuture.thenCompose(subspace ->
                         subdir.getValueRange(context, valueRange, subspace).thenApply(range -> {
+                            // The ChainedCursor is going to do counting of our reads to apply any limits
+                            // that were specified on the ScanProperties.  We don't want the inner
+                            // KeyValueCursor in nextTuple() to ALSO count those same reads so we clear
+                            // out its limits.
+                            final ScanProperties singleRowScan = scanProperties.with(
+                                    props -> props.clearState().setReturnedRowLimit(1));
```

Comment:
This raises an important question which I'm not sure how to answer. In addition to the time and record scan limits, the `CursorLimitManager` also enforces the byte scan limit (added in #349). It seems hard to make the `ChainedCursor` enforce that, though, since it doesn't actually see the key-value pairs that its underlying cursors are scanning. On the other hand, we don't have a mechanism here for telling the generated cursors to enforce the byte scan limit.

One solution here would be to require the `nextGenerator` to take a `ByteScanLimiter` or `CursorLimitManager` to increment. I'm not sure that I totally like that solution, though.

In any case, we should carefully document this contract before it confuses someone.

Reply:
I updated the java docs for `ChainedCursor` and opens issue #375 to address it.